clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1365,7 +1482,11 @@ public KafkaConsumerMetrics kafkaConsumerMetrics() {
     private void maybeThrowFencedInstanceException() {
         if (isFenced) {
             throw new FencedInstanceIdException("Get fenced exception for group.instance.id " +
-                groupInstanceId.orElse("null"));
+                groupMetadata.orElseThrow(
+                    () -> new IllegalStateException("No group metadata found although a group ID was provided. This is a bug!")
+                ).groupInstanceId().orElseThrow(
+                    () -> new IllegalStateException("No group instance ID found although the consumer is fenced. This is a bug!")
+                ));
```

Comment:
Can we restructure this so that our exception throwing case doesn't have nested exception throwing cases?